bluemira/magnets/strand.py

```diff
@@ -0,0 +1,421 @@
+            fig, ax = plt.subplots()
+
+        Ic_sc = self.Ic(B=B , T=T, T_margin=T_margin)
+        ax.plot(B, Ic_sc)
+        # Adding the plot title and axis labels
+        plt.title(f"Critical current for {self.__class__.__name__}\n"
+                  f"T = {T} [K], Tmargin = {T_margin} [K], strain = {strain}")  # Title
+        plt.xlabel('B [T]')            # X-axis label
+        plt.ylabel('Ic [A]')           # Y-axis label
+        # Enabling the grid
+        plt.grid(True)
+        if show:
+            plt.show()
+        return ax
+
+class WireNb3Sn(SuperconductingStrand):
+    """Represents an Nb3Sn strand made of 50% Copper100 and 50% Nb3Sn."""
+
+    # superconducting parameters for the calculation of Ic
+    c_ = 1.0
+    Ca1 = 50.06
+    Ca2 = 0.00
+    eps_0a = 0.00312
+    eps_m = -0.00059
+    Bc20max = 33.24
+    Tc0max = 16.34
+    p = 0.593
+    q = 2.156
+    C = 83075 * 1e6
+
+    def __init__(self, d_strand: float = 0.82e-3):
+        """
+        Initialize a WireNb3Sn instance.
+
+        Parameters
+        ----------
+        d_strand:
+            Strand diameter in meters. Default is 0.82e-3.
+        """
+        copper_100 = Copper100()
+        mat_Nb3Sn = Nb3Sn()  # noqa: N806
+        # materials: first material is the SC, then the other materials
+        materials = [mat_Nb3Sn, copper_100]
+        percentage = [0.5, 0.5]
+        super().__init__(materials=materials, percentage=percentage, d_strand=d_strand)
+
+    def Jc(
+            self, B: float, T: float, strain: float = 0.55, T_margin: float = 1.5,
+            **kwargs
+    ) -> float:
+        """
+        Returns the strand critical current density.
+
+        Parameters
+        ----------
+        B:
+            Operating magnetic field in Teslas.
+        T:
+            Operating temperature in Kelvins.
+        eps:
+            Total applied measured strain in percentage. Default is 0.55.
+        T_margin:
+            Strand temperature margin in operation in Kelvins. Default is 1.5.
+
+        Returns
+        -------
+        Strand current density in Amperes per square meter.
+        """
+        T_ = T + T_margin  # noqa: N806
+        int_eps = -strain / 100
+        eps_sh = self.Ca2 * self.eps_0a / (np.sqrt(self.Ca1 ** 2 - self.Ca2 ** 2))
+        s_eps = 1 + (
+                self.Ca1
+                * (
+                        np.sqrt(eps_sh ** 2 + self.eps_0a ** 2)
+                        - np.sqrt((int_eps - eps_sh) ** 2 + self.eps_0a ** 2)
+                )
+                - self.Ca2 * int_eps
+        ) / (1 - self.Ca1 * self.eps_0a)
+        Bc0_eps = self.Bc20max * s_eps
+        Tc0_eps = self.Tc0max * (s_eps) ** (1 / 3)  # noqa: N806
+        t = T_ / Tc0_eps
+        BcT_eps = Bc0_eps * (1 - t ** (1.52))
+        b = B / BcT_eps
+        hT = (1 - t ** (1.52)) * (1 - t ** 2)  # noqa: N806
+        fPb = (b ** self.p) * (1 - b) ** self.q  # noqa: N806
+        return self.c_ * (self.C / B) * s_eps * fPb * hT
```

Comment:
From my own implementations of these superconducting parameterisations, it is usually worthwhile to protect against NaNs and infinity. See PROCESS for an example of how to treat this.